listenstore/listenstore/listen.py

```diff
@@ -9,7 +9,7 @@ def __init__(self, user_id=None, timestamp=None, artist_msid=None, album_msid=No
                  recording_msid=None, data=None):
         self.user_id = user_id
         self.timestamp = timestamp
-        self.ts_since_epoch = calendar.timegm(self.timestamp.utctimetuple())
+        self.ts_since_epoch = calendar.timegm(self.timestamp.utctimetuple()) if self.timestamp else None
```

Comment:
What does this mean? Why can a timestamp be `None`? What does it mean to have no timestamp value in the database?